integration/Substra/plot_substrafl_flamby.py

```diff
@@ -0,0 +1,427 @@
+from substrafl.dependency import Dependency
+from substrafl.strategies import FedAvg
+from substrafl.nodes import AggregationNode
+from substrafl.evaluation_strategy import EvaluationStrategy
+from substrafl.index_generator import NpIndexGenerator
+from substrafl.experiment import execute_experiment
+
+import torch
+
+# %%
+# Substrafl algo definition
+# ==========================
+#
+# To instantiate a Substrafl :ref:`substrafl_doc/api/algorithms:Torch Algorithms`, you need to define a torch Dataset
+# with a specific `__init__` signature, that must contain (self, x, y, is_inference). This torch Dataset is useful to
+# preprocess your data on the `__getitem__` function.
+# The `__getitem__` function is expected to return x and y if is_inference is False, else x.
+# This behavior can be changed by re-writing the `_local_train` or `predict` methods.
+#
+# This dataset is passed **as a class** to the :ref:`substrafl_doc/api/algorithms:Torch Algorithms`.
+# Indeed, this torch Dataset will be instantiated within the algorithm, using the opener functions as x and y
+# parameters.
+#
+# The index generator will be used a the batch sampler of the dataset, in order to save the state of the seen samples
+# during the training, as Federated Algorithms have a notion of `num_updates`, which forced the batch sampler of the
+# dataset to be stateful.
+
+# Number of model update between each FL strategy aggregation.
+NUM_UPDATES = 16
+
+index_generator = NpIndexGenerator(
+    batch_size=batch_size,
+    num_updates=NUM_UPDATES,
+)
+
+
+class TorchDataset:
+    def __new__(self, x, y, is_inference):
+        return x
+
+    def __init__(self, x, y, is_inference):
+        pass
+
+
+class MyAlgo(TorchFedAvgAlgo):
+    def __init__(self):
+        super().__init__(
+            model=model,
+            criterion=criterion,
+            optimizer=optimizer,
+            index_generator=index_generator,
+            dataset=TorchDataset,
+            seed=42,
+        )
+
+    def _local_predict(self, predict_dataset: torch.utils.data.Dataset, predictions_path):
+
+        predict_loader = torch.utils.data.DataLoader(predict_dataset, batch_size=self._index_generator.batch_size)
+
+        self._model.eval()
+
+        predictions = torch.Tensor([])
+        with torch.inference_mode():
+            for x, _ in predict_loader:
+                x = x.to(self._device)
+                predictions = torch.cat((predictions, self._model(x)), 0)
+
+        predictions = predictions.cpu().detach()
+        self._save_predictions(predictions, predictions_path)
+
+        return predictions
+
+
+# %%
+# Algo dependencies
+# =================
+#
+# The **dependencies** needed for the :ref:`substrafl_doc/api/algorithms:Torch Algorithms` are specified by a
+# :ref:`substrafl_doc/api/dependency:Dependency` object, in order to install the right library in the Python
+# environment of each organization.
+
+algo_deps = Dependency(pypi_dependencies=["torch==1.12.0"])
```

Comment:
Why is it the only dep we need ?

Reply:
In fact the dependencies are not used in local mode. I added a comments about this. 
Indeed I'm not sure that this is the only dep we need, I'll check on my side !